scapy/sendrecv.py

```diff
@@ -338,7 +338,11 @@ def sendpfast(x, pps=None, mbps=None, realtime=None, loop=0, file_cache=False, i
     except KeyboardInterrupt:
         log_interactive.info("Interrupted by user")
     except Exception as e:
-        log_interactive.error("while trying to exec [%s]: %s", argv[0], e)
+        msg = "An error occured while trying to exec [%s]: %s" % (argv[0], e)
+        if conf.interactive:
+            log_interactive.error(msg)
```

Comment:
I'd prefer (occur**r**ed, `e` => `exc`, format done by the log function):
```python
    except Exception as exc:
        msg = "An error occurred while trying to exec [%s]%s"
        if conf.interactive:
            log_interactive.error(msg, argv[0], "", exc_info=True)
        else:
            raise Scapy_Exception(msg % (argv[0], ": %s" % exc))
```

But actually I don't think we need to raise a Scapy_Exception here, we can leave the Python exception, which might be easier to understand and work with for a developer:

```python
    except Exception:
        if conf.interactive:
            log_interactive.error("An error occurred while trying to exec [%s]",
                                                  argv[0], exc_info=True)
        else:
            raise
```

Reply:
In this specific case, the `Exception` message is not easy to understand. That's why I decided to raise a `Scapy_Exception`.

Reply:
I don't know. When one uses the Python API (and not Scapy's shell), he is a developer. So he expects "real" exceptions, not user-friendly messages. That's the whole point of raising an exception rather than displaying a message.

Reply:
Updated.